control-plane-operator/controllers/hostedcontrolplane/manifests/manifests.go

```diff
@@ -0,0 +1,146 @@
+	openshiftAPIServerServiceName = "openshift-apiserver"
+	oauthAPIServerName            = "openshift-oauth-apiserver"
+	DefaultAPIServerPort          = 6443
+	VPNServicePort                = 1194
+)
+
+func KubeAPIServerService(hostedClusterNamespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kubeAPIServerServiceName,
+			Namespace: hostedClusterNamespace,
+		},
+	}
+}
+
+func KubeAPIServerServicePorts(securePort int32) []corev1.ServicePort {
+	return []corev1.ServicePort{
+		{
+			Port:       securePort,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(int(securePort)),
+		},
+	}
+}
+
+func KubeAPIServerServiceSelector() map[string]string {
+	return map[string]string{"app": "kube-apiserver"}
+}
+
+func OauthServerService(hostedClusterNamespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OauthServiceName,
+			Namespace: hostedClusterNamespace,
+		},
+	}
+}
+
+func OauthServerServicePorts() []corev1.ServicePort {
+	return []corev1.ServicePort{
+		{
+			Name:       "https",
+			Port:       443,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(6443),
+		},
+	}
+}
+
+func OauthServerServiceSelector() map[string]string {
+	return map[string]string{"app": "oauth-openshift"}
+}
+
+func OauthServerRoute(hostedClusterNamespace string) *routev1.Route {
+	return &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: hostedClusterNamespace,
+			Name:      oauthRouteName,
+		},
+	}
+}
+
+func VPNServerService(hostedClusterNamespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vpnServiceName,
+			Namespace: hostedClusterNamespace,
+		},
+	}
+}
+
+func VPNServerServicePorts() []corev1.ServicePort {
+	return []corev1.ServicePort{
+		{
+			Port:       VPNServicePort,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(VPNServicePort),
+		},
+	}
+}
+
+func VPNServerServiceSelector() map[string]string {
+	return map[string]string{"app": "openvpn-server"}
+}
+
+func OpenshiftAPIServerService(hostedClusterNamespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      openshiftAPIServerServiceName,
+			Namespace: hostedClusterNamespace,
+		},
+	}
+}
+
+func OpenshiftAPIServerServicePorts() []corev1.ServicePort {
+	return []corev1.ServicePort{
+		{
+			Name:       "https",
+			Port:       443,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(8443),
+		},
+	}
+}
+
+func OpenshiftAPIServerServiceSelector() map[string]string {
+	return map[string]string{"app": "openshift-apiserver"}
+}
+
+func OauthAPIServerService(hostedClusterNamespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      oauthAPIServerName,
+			Namespace: hostedClusterNamespace,
+		},
+	}
+}
+
+func OauthAPIServerServicePorts() []corev1.ServicePort {
```

Comment:
As mentioned elsewhere, let's keep this package scoped to names only, not reconcile logic/data